_posts/2022-11-12-Aya-gsoc-final-report.md

```diff
@@ -0,0 +1,150 @@
+
+N.B.: it was important to have all the work on developing both the guesser API and the [DefaultGuesser] plus changing parsers behavior to be done inside one pull request here: [#3753]; because all thoses changes have a direct effect on each others, for example, the optmization of the new guesser API must be measured by how far the topology output changed after removing all the guessings processes from parsers. This was the most challenging thing about the project; as all these required changes are interconnected and must be done simultaneously, to see how it reflects and interacts with each other, and accordingly we can achieve optimization. So, working on this PR was not sequential as mentioned below, it was rather a continuous back-and-forth update on all touched parts of the library.
+
+#### - Removing types and mass guessing from Topology Parsers ([commit 0cbc497]) 
+First, I navigated through all parsers to see where and how types and mass guessing take place. The below table shows the behavior of types and masses guessing inside parsers.
+![Guessing inside Parsers](public/images/final_report_aya/parsers_guessing.png)
+As we see, not all parsers use the same attribute in guessing masses. The [DefaultGuesser] guesses masses by first looking for the elements attribute and if not available, it looks for the types attribute, this behavior preserves the default behavior of all parsers except the ones that guess masses from names. Parsers that guess masses from types are [FHIAIMSParser], [TXYZParser], and [XYZParser]. For both FHIAIMSParser and XYZParser `element` attributes are provided, and luckily it is just a copy of the `names` attribute, so the default behavior of these parsers is not affected. For TXYZParser, I added a feature of publishing Elements attribute to its topology output in case all read names are valid elements ([merged PR #3836]).
+
+#### - Developing the `guess_topologyAttributes` API
+As described in issue [# 3704], the new guesser API will have the tasks of guessing and adding new topology attributes to the [Universe]. The user only passes the context (eg. "default", "pdb", "martini") and the attributes of interest to be guessed, either through the `to_guess` or `force_guess` parameters and the API does the rest. Since I was writing my proposal, I was keen to develop the guesser API with this high level of abstraction and make guessing attributes easy and straightforward to the user without bothering about which method should be called and which attribute(s) is used as a reference in guessing other attributes, so all this work is handled inside the API and the guessers classes. The `guess_topologyAttributes()` make the following processes:
+
+a- gets the appropriate Guesser class that matches the passed context.
+
+b- checks if the Guesser class support guessing the attributes passed to the`to_guess` and/or `force_guess` parameters.
```

Comment:
I would just use the standard markdown formatting for lists
```markdown
1. gets the appropriate Guesser class that matches the passed context.

2. checks if the Guesser class support guessing the attributes passed to the`to_guess` and/or `force_guess` parameters.
```
and then it will look like this:

## Markdown list
![image](https://user-images.githubusercontent.com/237980/206595992-399fa203-3fff-4417-beb3-786314aa78bd.png)



## Current manually created list

Currently it looks like
![image](https://user-images.githubusercontent.com/237980/206595760-789617f3-8ece-46b0-8122-64eb0a276d69.png)